sample/src/main/java/io/mosaicnetworks/sample/ChatActivity.java

```diff
@@ -26,112 +26,88 @@
 
 import android.content.Context;
 import android.content.Intent;
-
-import androidx.annotation.StringRes;
-import androidx.appcompat.app.AppCompatActivity;
 import android.os.Bundle;
-import android.util.Log;
 import android.view.Menu;
 import android.view.MenuItem;
 import android.view.View;
-import android.widget.ImageView;
-import android.widget.LinearLayout;
-import android.widget.TextView;
-import android.widget.Toast;
 
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
-import com.squareup.picasso.Picasso;
 
 import java.text.SimpleDateFormat;
-import java.util.ArrayList;
 import java.util.Date;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 
 import io.mosaicnetworks.babble.discovery.Peer;
-import io.mosaicnetworks.babble.node.ServiceObserver;
+import io.mosaicnetworks.babble.node.BabbleNode;
+import io.mosaicnetworks.babble.service.BabbleServiceBinderActivity;
+import io.mosaicnetworks.babble.service.ServiceObserver2;
 import io.mosaicnetworks.babble.utils.DialogUtils;
 import io.mosaicnetworks.babble.utils.Utils;
-import io.mosaicnetworks.sample.chatkit.commons.ImageLoader;
 import io.mosaicnetworks.sample.chatkit.commons.models.IMessage;
+import io.mosaicnetworks.sample.chatkit.messages.MessageHolders;
 import io.mosaicnetworks.sample.chatkit.messages.MessageInput;
 import io.mosaicnetworks.sample.chatkit.messages.MessagesList;
 import io.mosaicnetworks.sample.chatkit.messages.MessagesListAdapter;
-import io.mosaicnetworks.sample.notification.NotificationMessage;
+import io.mosaicnetworks.sample.notification.Checker;
+import io.mosaicnetworks.sample.notification.NotificationHolder;
 
 /**
- * This is the central UI component. It receives messages from the {@link MessagingService} and
+ * This is the central UI component. It receives messages from the BabbleService} and
  * displays them as a list.
  */
-public class ChatActivity extends AppCompatActivity implements ServiceObserver, StatsObserver  {
+public class ChatActivity extends BabbleServiceBinderActivity implements ServiceObserver2 {
```

Comment:
We need an additional stage in this exercise. 

We have lots of class names ending in 2. Do we, after this PR, refactor them to lose the 2s where non-clashing. 

E.g. BabbleService2 -> BabbleService

Reply:
Agree. Given that the tools will help us here, it's worth me fixing this now